### PR TITLE
Refactoring to use profiles.tuned.openshift.io

### PR DESCRIPTION
**Pull request by jmencak**
This is a largish refactoring of the NTO.
Changes:
  - The logic for tuned profile calculation was moved from NTO's
    openshift-tuned operands to the operator itself.
  - ConfigMap that carried a list of recommended profiles was removed,
    profiles.tuned.openshift.io was introduced to carry the calculated
    per-node profile information.  tuned-profiles ConfigMap that
    carries a list of default and all user-defined tuned profiles is
    still being used.  This is likely to be replaced by an API object
    too -- future work.
  - Operand's SA/RBAC definitions were moved to operator manifests.
  - No dependency on controller-runtime.
  - Operand's DaemonSet adjustments to make certain fields more
    explicit.
  - Switch to klog, glog seems unmaintained and vendored-in libraries
    already depend on klog.
  - Makefile/.gitignore cleanup
  - Added vendor dependencies

### CODE REVIEW

**Comment by jmencak** [comment]
/cc @sjug , @Deepthidharwar 
/hold
This is a 4.4 branch code, do not merge yet.
Needs to merge together with https://github.com/openshift/openshift-tuned/pull/34

**Comment by zvonkok** [comment]
Looking good for me, with the changes in the operand https://github.com/openshift/openshift-tuned/pull/34 this PR will address all current outstanding issued with the operator.

File: pkg/operator/controller.go [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+// read and process a message on the workqueue.
+func (c *Controller) eventProcessor() {
+	for {
+		// Wait until there is a new item in the working queue
+		obj, shutdown := c.workqueue.Get()
+		if shutdown {
+			return
+		}
+
+		klog.V(2).Infof("get event from workqueue")
```
Comment (sjug) [inline pkg/operator/controller.go]: Only small logging change, should be "got event"

Comment (jmencak) [reply pkg/operator/controller.go]: Thanks, fixed!

**Comment by jmencak** [comment]
CI failing on 
```
level=info msg="API v1.16.2 up"
level=info msg="Waiting up to 30m0s for bootstrapping to complete..."
level=info msg="Destroying the bootstrap resources..."
level=info msg="Waiting up to 30m0s for the cluster at https://api.ci-op-10bmlm2z-ef33c.origin-ci-int-aws.dev.rhcloud.com:6443 to initialize..."
level=info msg="Cluster operator insights Disabled is False with : "
level=info msg="Cluster operator support Disabled is True with NotAuthorized: Reporting was not allowed: your Red Hat account is not enabled for remote support or your token has expired"
level=fatal msg="failed to initialize the cluster: Cluster operator node-tuning has not yet reported success"
```
@petr-muller any chance you know what's going on?  Thanks!

**Pull request merged**